report/3.Theory/SyntaxTheory.tex

```diff
@@ -0,0 +1,170 @@
+% This document needs to include:
+% - An introduction for the concern.
+% - Bases for section, specifically EBNF in previous section.
+% - Difference between EBNF CFG.
+% - HCL as CFL and HCL's CFG.
+% - Proof that HCL is a CFL.
+% - Proof that shows whether HCL is a regular language or not.
+% - Description of identifiers and types in HCL using computational models or Regex.
+% - Conclusion.
+\section{Mathematical Syntax Theory}
+This section concerns with the concise definitions of HCL's syntax.
+First the proper CFG recognizing HCL will be presented, which will be used to proof that HCL is a CFL. 
+Afterwards the most specific computational model for describing HCL will be identified and presented. 
+Lastly the token categories will be described in a computational model. 
+The conclusions of this section will provide the bases for implementing and testing the parser of the HCL compiler. 
+The majority of this section will be based on the EBNF details touched upon in the previous chapter.
+
+\subsection{HCL as an CFL}
+This subsection will investigate whether HCL is a CFL. 
+To do this it is necessary to construct a PDA.
+The reason being that the set of context free languages is equivalent to the set of languages recognized by push down automata.
```

Comment:
har skrevet den kommentar her før...
når vi har abbreviations for PDA og CFL, skal de bruges <({[HVER GANG]})>

Reply:
Logikken var:
CFL -> Context free language
ikke: context free languages
Men hvis vi kan skrive CFL's og det er  fint så gør vi bare det

Det er blevet fikset i et commit så er alt er godt nu :)